### PR TITLE
[jmxfetch] add rmi registry ssl config option

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Add new config options introduced by https://github.com/DataDog/jmxfetch/pull/185

### CODE REVIEW

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 1]
```diff
@@ -20,11 +20,14 @@ instances:
   #   tools_jar_path: /usr/lib/jvm/java-7-openjdk-amd64/lib/tools.jar # To be set when process_name_regex is set
 
   #   name: jmx_instance
-  #   # java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
-  #   # java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
-  #   # trust_store_path: /path/to/trustStore.jks # Optional, should be set if ssl is enabled
-  #   # trust_store_password: password
-  #   # refresh_beans: 600 # Optional (in seconds), default is 600 seconds. Sets refresh period for refreshing matching MBeans list.
```
Comment (olivielpeau) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: the double `#` here were meaningful to reflect that the options are optional. Just pointing that out, not sure it should make sense to keep things this way

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 2]
```diff
@@ -20,11 +20,14 @@ instances:
...
+  #   java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
+  #   java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
+  #   trust_store_path: /path/to/trustStore.jks # Optional, should be set if "com.sun.management.jmxremote.ssl" is set to true.
```
Comment (olivielpeau) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: `is set to true` on the target JVM I assume?

may require a longer explanation on what this `jks` file is exactly (not necessarily here though, full explanation could be on docs website)

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 3]
```diff
@@ -20,11 +20,14 @@ instances:
...
+  #   trust_store_password: password
+  #   key_store_path: /path/to/keyStore.jks # Optional, should be set if "com.sun.management.jmxremote.ssl.need.client.auth" is set to true.
+  #   key_store_password: password
+  #   rmi_registry_ssl: false # Optional, should be set to true if "com.sun.management.jmxremote.registry.ssl" is set to true.
+  #   refresh_beans: 600 # Optional (in seconds), default is 600 seconds. Sets refresh period for refreshing matching MBeans list.
   #                        # Decreasing this value may result in increased CPU usage.
```
Comment (olivielpeau) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: inconsistent indentation

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 4]
```diff
@@ -20,11 +20,14 @@ instances:
...
-  #   # java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
-  #   # java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
-  #   # trust_store_path: /path/to/trustStore.jks # Optional, should be set if ssl is enabled
-  #   # trust_store_password: password
-  #   # refresh_beans: 600 # Optional (in seconds), default is 600 seconds. Sets refresh period for refreshing matching MBeans list.
+  #   java_bin_path: /path/to/java # Optional, should be set if the agent cannot find your java executable
+  #   java_options: "-Xmx200m -Xms50m" # Optional, Java JVM options
+  #   trust_store_path: /path/to/trustStore.jks # Optional, should be set if "com.sun.management.jmxremote.ssl" is set to true.
+  #   trust_store_password: password
+  #   key_store_path: /path/to/keyStore.jks # Optional, should be set if "com.sun.management.jmxremote.ssl.need.client.auth" is set to true.
```
Comment (olivielpeau) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: same comments as for `trust_store_path `

**Review by olivielpeau — COMMENTED** [review]
Thanks!

Left a few comments, let me know what you think.

Also, we shouldn't merge this until the Agent actually uses a new version of JMXFetch that supports these options

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example (hunk 1 above)
Comment (arbll) [reply cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: Only `host` and `port` are really non-optional and we don't do this in most configuration so it looks inconsistent to me.

**Review by olivielpeau — DISMISSED** [review]
just an indentation nit, other than that LGTM 👍

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**